spec/vector_spec.rb

```diff
@@ -254,6 +254,33 @@
             expect { @vector[:x, :one] }.to raise_error(IndexError)
           end
         end
+
+        context Daru::CategoricalIndex do
+          let(:idx1) { Daru::CategoricalIndex.new [:a, :b, :a, :a, :c] }
+          let (:dv1) { Daru::Vector.new 'a'..'e', index: @idx1 }
+
+          it "returns vector of elements belonging to given category" do
+            expect(@dv1[:a]).to eq(Daru::Vector.new(
+              ['a', 'c', 'd'], index: Daru::CategoricalIndex(
+                [:a, :a, :a])))
+          end
+
+          it "returns vector of elements belonging to given categories" do
+            expect(@dv1[:a, :c]).to eq(Daru::Vector.new(
+              ['a', 'c', 'd', 'e'], index: Daru::CategoricalIndex.new(
+                [:a, :a, :a, :c])))
```

Comment:
> Though, I should say that such things are mostly the matter of personal preferences, so it is just a suggestion to consider. I will not bother you with those things too frequent :)

I have very less experience with this. Please let me know of any suggestion you have :)